### PR TITLE
Tweaks

### PR DESCRIPTION
**Pull request by ThierryO**
I'v reworked `render_release()`.

Solves #18 
Solves #19 
Solves #15 

- apply INBO css styling
- add unit test

### CODE REVIEW

**Comment by ThierryO** [comment]
I've used this folder structure.

- specific version of a proctocol is located at `publish/<YYYY.XX>/index.html`. The user will find them at e.g. https://protocol.inbo.be/2020.01. Every protocol version links to `publish/<protocol_code>/index.html`
- all versions of a proctocol (with links) are listed at `publish/protocol_code/index.html`. The user finds it at e.g. https://protocol.inbo.be/sfp-201
- The complete list of all protocols, and their versions, is appended to `src/homepage.Rmd` and rendered to `publish/index.html`. The user finds it at https://protocol.inbo.be

**Comment by florisvdh** [comment]
While leaving full review to @hansvancalster, just a few superficial remarks based on your description:

- > specific version of a proctocol is located at `publish/<YYYY.XX>/index.html`

  protocols can be available in > 1 language; does this directory structure allow that? Probably the language tag like `nl` / `en` should be at some level as well? (for the website probably at a top level)
- I suggest to use protocol**s**.inbo.be rather than protocol.inbo.be

**Comment by florisvdh** [comment]
@ThierryO Having run your version with the script of @hansvancalster (without the argument in `render_release()`), this works nicely. :+1: 

Questions / suggestions:

- can clashes occur between the manually inserted version number in NEWS.Rmd and the one, generated here? Is it checked whether the top version number in NEWS.Rmd of newly released protocols matches the one implemented by `render_release()`?
- related to the above: I did not look into the code to see what the actual approach is. I suggest to clarify it in the documentation of `render_release()`.

A few remarks on the implemented layout (I guess this relates to INBOmd):

- body text is too small IMHO:

  image

- in the above screenshot, I suggest to make the contrast (for body text) larger. I'd suggest to drop the background.

- in the gitbook, I suggest avoid the appearance of a white text box surrounded by a grey background (rather drop the background as in https://r4ds.had.co.nz/tibbles.html):

image

It is a great move that we can implement an INBO corporate style! While the latter may need further tampering (e.g. above, but more opinions are needed), I feel that the `protocolhelper` package should not enforce our corporate style because the philosophy is also the usability for other organisations, AFAIK. So perhaps that could be managed with an argument.

**Comment by florisvdh** [comment]
I noticed a few more things in the results, when using https://github.com/hansvancalster/protocols/blob/master/src/management/setup_dummies.R:

- in `publish/index.html` :
    - the sorting of headers should follow [protocol type and theme number](https://github.com/inbo/protocols#protocol-code-and-protocol-number-)
    - the English version of sfp-001 is still missing
- relating to INBO corporate style: hyperlinks in `publish/index.html` (links to _protocol_code_ and _version_number_) and in each protocol-specific `index.html` (link to _version_number_): links are invisible unless when one hovers above them

**Comment by ThierryO** [comment]
> * can clashes occur between the manually inserted version number in NEWS.Rmd and the one, generated here? Is it checked whether the top version number in NEWS.Rmd of newly released protocols matches the one implemented by `render_release()`?

The version of a protocol is based on the version number mentioned in the yaml of the protocol. These version numbers are used on the homepage. The overview page of the protocol is based on `NEWS.Rmd` and needs manual updating. Human error can introduce a mismatch.

> * related to the above: I did not look into the code to see what the actual approach is. I suggest to clarify it in the documentation of `render_release()`.

1. render all protocols were the latest version does not exists in `publish`. The most recent protocol (largest version number) is always rendered.
1. render the `NEWS.Rmd` without the heading as the protocol overview page.
1. list all protocol versions in `publish` and append them to the homepage
 
> A few remarks on the implemented layout (I guess this relates to INBOmd):
> 
> * body text is too small IMHO
> * in the gitbook, I suggest to avoid the appearance of a white text box surrounded by a grey background (rather drop the background as in https://r4ds.had.co.nz/tibbles.html):

It is based on the current INBO website and used in INBOmd. A new version of the website will be launched by the end of the year. I don't want to waiste time on fixing css that will be replaced soon.
Note that increasing the font size is an easy thing to do in your browser.

**Comment by ThierryO** [comment]
> I noticed a few more things in the results, when using https://github.com/hansvancalster/protocols/blob/master/src/management/setup_dummies.R:

I've setup a dummy repo using https://github.com/inbo/protocolhelper/blob/tweaks/tests/testthat/test-render_release.R

File: R/render_release.R [hunk 1]
```diff
@@ -35,109 +35,178 @@
...
+    regexp = "[0-9]{4}\\.[0-9]{2}/index.html"
+  )
+  meta <- map(
+    protocols,
+    function(x) {
+      meta <- readLines(x, n = 40)
+      meta <- str_subset(meta, pattern = "<meta name=\"protocol")
+      z <- as.list(str_replace(meta, ".*content=\"(.*?)\".*", "\\1"))
+      names(z) <- str_replace(meta, ".*name=\"protocol-(.*?)\".*", "\\1")
+      as.data.frame(z)
```
Comment (hansvancalster) [inline R/render_release.R:169]: I'm having a hard time seeing what this code does.... can you explain a bit?

File: R/render_release.R [hunk 2]
```diff
@@ -35,109 +35,178 @@
...
+            "protocol_code", parameters[[i]][["protocol_code"]]
+          ),
+          pandoc_variable_arg(
+            "version_number", parameters[[i]][["version_number"]]
+          ),
+          pandoc_variable_arg("thema", parameters[[i]][["theme"]]),
+          pandoc_variable_arg("lang", parameters[[i]][["language"]])
+        ),
+        template = "css/gitbook.html"
+      ),
```
Comment (hansvancalster) [inline R/render_release.R:119]: Is this necessary? Except for line 118, I think it is already defined in _bookdown.yml and yaml front matter in the index.Rmd files.

File: inst/rmarkdown/templates/template_sfp_en/skeleton/skeleton.Rmd [hunk 3]
```diff
@@ -45,21 +45,10 @@ opts_chunk$set(
 )
 ```
 
+| Reviewers | Documentbeheerder | Protocol code | Versienummer | Taal | Thema |
+| --------- | ----------------- | ------------- | ------------ | ---- | ----- |
+| `r paste(params$reviewers, collapse = ", ")` | `r params$file_manager` | `r sprintf("[%1$s](../%1$s/index.html)", params$protocol_code)` | `r params$version_number` | `r params$language` | `r params$theme` |
```
Comment (hansvancalster) [inline inst/rmarkdown/templates/template_sfp_en/skeleton/skeleton.Rmd:50]:
```suggestion
| Reviewers | File manager | Protocol code | Version number | Language | Theme |
| --------- | ----------------- | ------------- | ------------ | ---- | ----- |
| `r paste(params$reviewers, collapse = ", ")` | `r params$file_manager` | `r sprintf("[%1$s](../%1$s/index.html)", params$protocol_code)` | `r params$version_number` | `r params$language` | `r params$theme` |
```

**Comment by hansvancalster** [comment]
@ThierryO forgot to mention the test you wrote for `render_release()`, which was also a very nice addition. I tried it and it worked without a problem!

File: R/render_release.R (hunk 2 above)
Comment (ThierryO) [reply R/render_release.R:119]: Yes. This [line](https://github.com/inbo/protocolhelper/blob/882952bf707d3650abcd287719c13f25fe89a03b/R/render_release.R#L58) reads the information the index.Rmd. The `pandoc_variable_arg` stuff provides the information to the [template](https://github.com/inbo/protocolhelper/blob/882952bf707d3650abcd287719c13f25fe89a03b/inst/css/gitbook.html#L5), so it becomes metadata in the HTML. Which is reuse [later](https://github.com/inbo/protocolhelper/blob/882952bf707d3650abcd287719c13f25fe89a03b/R/render_release.R#L166)

File: R/render_release.R (hunk 1 above)
Comment (ThierryO) [reply R/render_release.R:169]: We've stored the meta information of protocols in the HTML through the [template](https://github.com/inbo/protocolhelper/blob/882952bf707d3650abcd287719c13f25fe89a03b/inst/css/gitbook.html#L5). This [section](https://github.com/inbo/protocolhelper/blob/882952bf707d3650abcd287719c13f25fe89a03b/R/render_release.R#L166) extracts that information from the individual HTML files. And then we combine it into an overview of the available protocols (and versions). Which we append to the homepage template.

File: R/render_release.R (hunk 2 above)
Comment (hansvancalster) [reply R/render_release.R:119]: Nice!

**Pull request merged**